tests/h_item/h_item_api.cpp

```diff
@@ -19,26 +23,78 @@ struct h_item_api_kernel_common;
  * @tparam currentDim Current dimension to get the value for
  */
 template <int numDims, int currentDim>
-struct h_item_api_kernel_single {
+struct kernel_single {
+  using success_acc_t =
+      cl::sycl::accessor<bool, 1, cl::sycl::access::mode::write>;
+
   cl::sycl::range<numDims> kernelLogicalLocalRange;
+  success_acc_t success_acc;
 
   void operator()(cl::sycl::group<numDims> group) const {
     group.parallel_for_work_item(
         kernelLogicalLocalRange, [&](cl::sycl::h_item<numDims> item) {
-          size_t globalRange = item.get_global_range(currentDim);
-          size_t globalId = item.get_global_id(currentDim);
-          size_t localRange = item.get_local_range(currentDim);
-          size_t localId = item.get_local_id(currentDim);
-          size_t logicalLocalRange = item.get_logical_local_range(currentDim);
-          size_t logicalLocalId = item.get_logical_local_id(currentDim);
-          size_t physicalLocalRange = item.get_physical_local_range(currentDim);
-          size_t physicalLocalId = item.get_physical_local_id(currentDim);
+          bool success = true;
+          {
+            auto value = item.get_global_range(currentDim);
+            auto expected = item.get_global_range()[currentDim];
+            success &= value == expected;
+          }
+          {
+            auto value = item.get_global_id(currentDim);
+            auto expected = item.get_global_id()[currentDim];
+            success &= value == expected;
+          }
+          {
+            auto value = item.get_local_range(currentDim);
+            auto expected = item.get_local_range()[currentDim];
+            success &= value == expected;
+          }
+          {
+            auto value = item.get_local_id(currentDim);
+            auto expected = item.get_local_id()[currentDim];
+            success &= value == expected;
+          }
+          {
+            auto value = item.get_logical_local_range(currentDim);
+            auto expected = item.get_logical_local_range()[currentDim];
+            success &= value == expected;
+          }
+          {
+            auto value = item.get_logical_local_id(currentDim);
+            auto expected = item.get_logical_local_id()[currentDim];
+            success &= value == expected;
+          }
+          {
+            auto value = item.get_physical_local_range(currentDim);
+            auto expected = item.get_physical_local_range()[currentDim];
+            success &= value == expected;
+          }
+          {
+            auto value = item.get_physical_local_id(currentDim);
+            auto expected = item.get_physical_local_id()[currentDim];
+            success &= value == expected;
+          }
+
+          if (!success) success_acc[0] = false;
         });
   }
 };
 
-namespace TEST_NAME {
-using namespace sycl_cts;
+template <template <int> class T, int numDims>
```

Comment:
I have the feeling that these functions could go into a global utility part of the infrastructure, if not existing already/

Reply:
I've removed these functions in bde9441